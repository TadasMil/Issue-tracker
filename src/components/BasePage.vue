<script setup lang="ts" name="BasePage">
const { title } = defineProps<{
  title: string
}>()
</script>

<template>
  <div class="m-4">
    <h2 class="mb-3">{{ title }}</h2>

    <main>
      <slot />
    </main>
  </div>
</template>
