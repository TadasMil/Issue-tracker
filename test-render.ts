import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
