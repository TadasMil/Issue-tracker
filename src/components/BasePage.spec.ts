import { describe, it, expect } from 'vitest'
import { render } from '../../test-render'

import BasePage from './BasePage.vue'

describe('BasePage', () => {
  it('should render the component given home page is active', () => {
    const { html } = render(BasePage, {
      props: {
        title: 'Home',
      },
    })

    expect(html()).toMatchSnapshot()
  })

  it('should render default slot', () => {
    const { html } = render(BasePage, {
      props: {
        title: 'Home',
      },
      slots: {
        default: 'Hello World',
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
