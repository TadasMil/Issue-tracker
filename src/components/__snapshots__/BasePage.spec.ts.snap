// Vitest Snapshot v1

exports[`BasePage > should render default slot 1`] = `
"<div class=\\"m-4\\">
  <h2 class=\\"mb-3\\">Home</h2>
  <main>Hello World</main>
</div>"
`;

exports[`BasePage > should render the component given home page is active 1`] = `
"<div class=\\"m-4\\">
  <h2 class=\\"mb-3\\">Home</h2>
  <main></main>
</div>"
`;
